daemon/graphdriver/overlay2/overlay.go

```diff
@@ -150,11 +161,33 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 		ctr:     graphdriver.NewRefCounter(graphdriver.NewFsChecker(graphdriver.FsMagicOverlay)),
 	}
 
-	return d, nil
-}
+	if backingFs == "xfs" {
+		// Try to enable project quota support over xfs.
+		if d.quotaCtl, err = graphdriver.NewQuotaCtl(home); err == nil {
+			//
+			// Test if file system supports project quotas by trying to set a quota
```

Comment:
I understand your concern. scan should be just as fast as ls -l /var/lib/docker/overlay2 > /dev/null.
but anyway, I pushed a change as you requested.